src/main/java/seedu/address/logic/commands/RemoveTagCommand.java

```
@@ -0,0 +1,68 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.tag.exceptions.TagNotFoundException;
+
+/**
+ * Removes a tag from all persons in the address book.
+ */
+public class RemoveTagCommand extends UndoableCommand {
+
+    public static final String COMMAND_WORD = "removetag";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Removes the tag from all persons.\n"
+            + "Parameters: TAG (must be an existing tag)\n"
+            + "Example: " + COMMAND_WORD + " subcommittee";
+
+    public static final String MESSAGE_REMOVE_TAG_SUCCESS = "Tag Removed: %1$s";
+
+    private Tag tagToRemove;
+
+    public RemoveTagCommand(Tag tagToRemove) {
+        this.tagToRemove = tagToRemove;
+    }
+
+    @Override
+    public CommandResult executeUndoableCommand() {
+        requireNonNull(tagToRemove);
+
+        try {
+            model.removeTag(tagToRemove);
+        } catch (TagNotFoundException tnfe) {
+            throw new AssertionError("The tag to be deleted cannot be missing");
+        }
+
+        return new CommandResult((String.format(MESSAGE_REMOVE_TAG_SUCCESS, tagToRemove)));
+    }
+
+    @Override
+    protected void preprocessUndoableCommand() throws CommandException {
```

This should not be here, `Logic` component should not have direct access to tags.
`Logic` is responsible for crafting the query and `Model` is responsible to processing the query.

If there is a need to check if a `Tag` exist, perform it in the `Model`. Suppose it doesn't, throw the exception there and let it propagate up.